representation/instance.metta

```diff
@@ -12,29 +13,72 @@
 (: InstanceSet (-> $score Type))
 (: mkSInstSet (-> (List (ScoredInstance $score)) (InstanceSet $score)))
 
-;; sortDeme -- sort scored instances in a deme by the composite score in a decreasing order
-;; this is a wrapper for the actual sorting function sortSInst
+;; --- Conversion helpers ---
+(: instToExpr (-> Instance Expression))
+(= (instToExpr (mkInst $lst)) (List.listToExpr $lst))
+
+(: exprToInst (-> Expression Instance))
+(= (exprToInst $expr) (mkInst (exprToList $expr)))
+
+(: sInstToExpr (-> (ScoredInstance $score) Expression))
+(= (sInstToExpr (mkSInst (mkPair $inst $score)))
+    (let* (
+      ($instExpr (instToExpr $inst))
+      ($scoreExpr (cScoreToExpr $score))
+    )
+      (cons-atom $instExpr $scoreExpr)))
+
+(= (exprToSInst $expr)
+    (let* (
+      (($instExpr $scoreExpr) (decons-atom $expr))
+      ($inst (exprToInst $instExpr))
+      ($score (exprToCScore $scoreExpr))
+    )
+      (mkSInst (mkPair $inst $score))))
+
+(= (sInstSetToExpr (mkSInstSet $lst)) (sInstSetListToExpr $lst))
```

Comment:
If you add an example here as a comment, it would simplify reading the nested function calls just to understand what kind of output we're expecting. Ofcourse adding type is the first thing to do but even that won't be enough as the return type will be `Expression` and we don't know what it looks like.

Reply:
@Biruk-gebru, you didn't add the example I mentioned